Gems/ROS2/Code/Source/RobotImporter/URDF/URDFPrefabMaker.cpp

```diff
@@ -177,6 +178,7 @@ namespace ROS2
 
         // Create the joints
         auto joints = Utils::GetAllJoints(m_model->root_link_->child_links);
+        AZStd::unordered_map<AZ::Name, AZ::EntityId> hierarchyMap;
```

Comment:
It seems that all we need from this hierarchy is the joint name, which we do not persist in any way during the import. Which is better to be added to the Joint component itself as a feature. The second best solution in my opinion is to add this extra information to ROS2Frame or even add a Component with extra information, but contained within the entity - and the hierarchy would only be acquired during Activate by the Component needing it.

Reply:
@adamdbrw We considered adding names to the joints themselves as well and @danielemorra98 actually created a corresponding PR https://github.com/o3de/o3de/pull/14195 but concluded that joint naming is not really a feature needed by many applications and thus extending the PhysX Gem itself not necessarily justified. Maintaining this metadata in ROS2Frame or a separate "metadata" component sounds reasonable though

Reply:
Indeed the decision about including Joint name affects all the users, and I am happy this was discussed. It would be best in term of encapsulation, but the name in Joint could be a "noise" field for many other users. Perhaps ROS2Frame would be suitable for an extension. If the "metadata" would outgrow or diminish intuition about what ROS2Frame is responsible for, a dedicated component (per entity) is nice as well.